### PR TITLE
Move to latest go version 1.18

### PR DESCRIPTION
**Pull request by gparvin**
Refs:
 - https://github.com/stolostron/backlog/issues/22476

Signed-off-by: Gus Parvin <gparvin@redhat.com>

### CODE REVIEW

**Review by dhaiducek — CHANGES_REQUESTED** [review]
The workflows also need to be updated to 1.18

**Review by mprahl — DISMISSED** [review]

**Comment by mprahl** [comment]
Since the required tests have changed to 1.18 to enable this PR, it has blocked #29.

**Comment by gparvin** [comment]
Actions are 1.18 now.
/unhold

File: build/common/config/.golangci.yml [hunk 1]
```diff
@@ -68,7 +78,8 @@ linters-settings:
     # report about shadowed variables
     check-shadowing: false
   gci:
-    local-prefixes: open-cluster-management.io/governance-policy-status-sync
+    sections:
+      - prefix(open-cluster-management.io/config-policy-controller)
```
Comment (dhaiducek) [inline build/common/config/.golangci.yml]: This doesn't match this repo. Is that intended?

Comment (gparvin) [reply build/common/config/.golangci.yml]: Good catch.  Will add a fix

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**